### PR TITLE
Document that weak-model semantic review needs a narrowed question and lenient structured-output parsing

### LINKED ISSUE #1552: Document that weak-model semantic review needs a narrowed question and lenient structured-output parsing
## Background

Parent roadmap: #1540 (weak-local-field-notes-r2).

An independent field PoC found that wherever a weak model is asked an open-ended semantic-quality question, or asked to emit a fixed structured-output line, the naive form is unreliable in specific, actionable ways: <https://gist.github.com/kurone-kito/8f45ee1ec0c07e8a7fdc8c1f04e83654>.

Evidence: (a) a single weak reviewer produced false, over-strict concerns on correct code (for example, claiming a passing reversal test "doesn't test reversal"). (b) A **majority panel of 3+ weak reviewers did not fix this** — on labeled test cases, all reviewers returned "concern" even on good units (both single-reviewer and majority-vote specificity measured 2/4), because the bias is **correlated and systematic** (a general "could test more" reflex), not independent per-reviewer noise that voting could cancel out. The lever that worked was **narrowing the question itself**: replacing an open-ended "is this good / does it cover edge cases?" with a narrow, falsifiable check — "does the verify contradict the spec? do not complain about coverage" — restored specificity to 4/4 even with a single reviewer. (c) Separately, a weak model asked to emit a fixed structured line (for example `UNIT name | target | contract`) frequently drifted into numbered lists, Markdown, code fences, or a dropped keyword, so a strict parser expecting the exact format returned nothing and aborted.

This is general guidance about the reliability shape of weak-model semantic judgment and structured-output emission — it does not propose changing any existing gate's scope or hard/advisory status. In particular, it does not reopen #909/#1342's decided-and-shipped scope for Copilot advisory-review convergence (a different question: whether an *external bot's* review blocks merge), and it does not by itself change `idd-review-triage.instructions.md` (E4-E8)'s exclusion from the lite execution profile recorded in `docs/weak-model-lite-profile-design.md` (a judgment-heavy phase this repository's design intentionally keeps off the confined weak-model tier) — the PoC's narrow-rubric E4 result is recorded here as a nuance worth a maintainer's future look, not as grounds to reopen that exclusion in this doc-only change.

## Proposed change

In "Weak-model guardrails" (or a new subsection near it), document two general principles for any weak-model semantic-judgment or structured-output step this repository's instructions or an adopter's own tooling may use:

1. **Narrow the question before adding judges.** An open-ended quality judgment from a weak model is unreliable even under majority voting, because the failure mode is a correlated bias rather than independent noise. A narrow, falsifiable question (for example, a direct contradiction check rather than an open-ended quality assessment) is far more usable, and should be preferred as the first lever — before reaching for a multi-reviewer panel — for any weak-model review step. Any such review output should be treated as advisory (a signal that can prompt a redraft), not a hard veto, consistent with how this repository already treats model-attested review elsewhere.
2. **Parse structured output leniently, with a bounded retry.** For any weak-model structured-output boundary (a fixed-format line, a marker, a short schema), treat drift as expected rather than exceptional: strip Markdown fences/list-markers/bold before matching, make secondary keywords optional, require at least one discriminating field to reject prose/headers, normalize and de-duplicate results, and provide a one-shot few-shot example in the prompt plus a bounded retry before treating the output as failed. State that this complements, not replaces, this repository's existing byte-exact marker verification (#1349): lenient parsing applies to a step that reads a weak model's free-form structured attempt, not to the mechanical audit of an already-published issue body's markers.

Add a one-line cross-reference from this new guidance to the E9-E15 lite-instruction file's own upstream-triage-boundary statement (see the "ship the condensed execution-loop instruction profile" roadmap), noting the E4 narrow-rubric finding as a recorded nuance for future consideration rather than a scope change made by this issue.

## Acceptance criteria

- [ ] "Weak-model guardrails" documents the narrow-the-question-before-adding-judges principle, with the correlated-bias-versus-independent-noise distinction stated explicitly.
- [ ] "Weak-model guardrails" documents the lenient-parsing-plus-bounded-retry principle for structured-output boundaries, and explicitly distinguishes it from the existing byte-exact marker verification.
- [ ] The addition explicitly states it does not change #909/#1342's scope or `idd-review-triage.instructions.md`'s lite-profile exclusion.
- [ ] `node scripts/audit-docs.mjs --check` passes.

---

_Autopilot suitability: 4 / 5 -- higher is more autopilot-suitable; below the configured floor is human-oriented._

_Effort: S -- author-estimated size; a soft autopilot selection tie-breaker only._

### PR DESCRIPTION
**Pull request by kurone-kito**
# Summary

Refines the weak-model guidance in both workflow docs.

## Linked issue

Closes #1552

## Follow-up issues (if any)

- [x] none

## Background / rationale (if material)

Documents the intended weak-model guardrails for narrow semantic judging,
bounded parsing retries, and the lite-profile roadmap boundary in both the live
and template workflow docs.

## IDD impact

- [x] Instruction files changed
- [x] Template files changed
- [ ] Helper scripts changed
- [ ] Config schema changed
- [ ] Security / credential / merge behavior changed

## Verification

- [x] `git diff --check`
- [x] `npx dprint check docs/idd-workflow.md idd-template/docs/idd-workflow.md`
- [x] `npx markdownlint-cli2 docs/idd-workflow.md idd-template/docs/idd-workflow.md`
- [x] `npx cspell lint docs/idd-workflow.md idd-template/docs/idd-workflow.md --no-progress`
- [x] `npx biome check && npx dprint check '**/*.md' && npx markdownlint-cli2 '**/*.md' && npx cspell lint '**' --no-progress && node scripts/idd-doctor.mjs`

## Safety

- [x] No secret-bearing examples
- [x] No private repo names / URLs
- [x] Merge policy impact reviewed
- [x] Context budget impact reviewed

## Summary by CodeRabbit

* **Documentation**
  * Expanded weak-model operating guidance with narrower, verifiable review checks.
  * Added bounded retry and lenient parsing guidance for structured results.
  * Clarified future roadmap boundaries and reaffirmed that lightweight models do not perform autonomous merge phases.

### CODE REVIEW

**Comment by kurone-kito** [comment]
_copilot-cli-8111acbc: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
**Accepted** — the Copilot review correctly identified the template ambiguity, and the fix now makes the upstream reference explicit in 15d01ff4.

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai did not review HEAD 15d01ff4 (review limit reached); this is not a completed review.

**Comment by kurone-kito** [comment]
**Rejected** — chatgpt-codex-connector did not review HEAD 15d01ff4 (usage limits reached); this is not a completed review.

File: idd-template/docs/idd-workflow.md [hunk 1]
```diff
@@ -122,6 +122,23 @@ When a lightweight-tier model runs any part of this loop:
...
+  discriminating field, normalize and de-duplicate the results, and
+  retry once with a short example before treating the attempt as failed.
+  This applies only to weak-model output readers and does not change the
+  byte-exact marker verification used for published IDD markers.
+- See the repository's lite-profile roadmap/design note for the E9-E15
+  upstream-triage boundary; the E4 narrow-rubric result is recorded
+  there as future nuance, not as a scope change here. This note does not
+  change the repository's Copilot advisory-review convergence scope
+  decision or the lite-profile exclusion of
+  `idd-review-triage.instructions.md`.
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/docs/idd-workflow.md]: **Accepted** — fixed in 15d01ff4: made the template reference explicitly point to the upstream IDD repository.

**Comment by kurone-kito** [comment]
advisory-wait: copilot-cli-8111acbc 15d01ff4fb240df6d08d3c8882510c319ffd6ab7 2026-07-19T18:23:09Z

**Comment by kurone-kito** [comment]
_copilot-cli-8111acbc: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
advisory-wait: copilot-cli-8111acbc fb69a5503da576c2b874e4c1afca1433db82d54d 2026-07-19T18:32:08Z

**Comment by kurone-kito** [comment]
advisory-wait: copilot-cli-8111acbc 647ba611284548e07717905150860dbd989532f6 2026-07-20T03:52:30Z
